flows/actions/call_classifier.go

```diff
@@ -0,0 +1,130 @@
+//     "type": "call_classifier",
+//     "classifier": {
+//       "uuid": "1c06c884-39dd-4ce4-ad9f-9a01cbe6c000",
+//       "name": "Booking"
+//     },
+//     "input": "@input.text",
+//     "result_name": "Intent"
+//   }
+//
+// @action call_classifier
+type CallClassifierAction struct {
+	baseAction
+	onlineAction
+
+	Classifier *assets.ClassifierReference `json:"classifier" validate:"required"`
+	Input      string                      `json:"input" validate:"required" engine:"evaluated"`
+	ResultName string                      `json:"result_name" validate:"required"`
+}
+
+// NewCallClassifier creates a new call classifier action
+func NewCallClassifier(uuid flows.ActionUUID, classifier *assets.ClassifierReference, input string, resultName string) *CallClassifierAction {
+	return &CallClassifierAction{
+		baseAction: newBaseAction(TypeCallClassifier, uuid),
+		Classifier: classifier,
+		Input:      input,
+		ResultName: resultName,
+	}
+}
+
+// Execute runs this action
+func (a *CallClassifierAction) Execute(run flows.FlowRun, step flows.Step, logModifier flows.ModifierCallback, logEvent flows.EventCallback) error {
+	classifiers := run.Session().Assets().Classifiers()
+	classifier := classifiers.Get(a.Classifier.UUID)
+
+	// substitute any variables in our input
+	input, err := run.EvaluateTemplate(a.Input)
+	if err != nil {
+		logEvent(events.NewError(err))
+	}
+
+	classification, skipped, err := a.classify(run, step, input, classifier, logEvent)
+	if err != nil {
+		logEvent(events.NewError(err))
+
+		if skipped {
+			a.saveSkipped(run, step, input, logEvent)
+		} else {
+			a.saveFailure(run, step, input, logEvent)
+		}
+	} else {
+		a.saveSuccess(run, step, input, classification, logEvent)
+	}
+
+	return nil
+}
+
+func (a *CallClassifierAction) classify(run flows.FlowRun, step flows.Step, input string, classifier *flows.Classifier, logEvent flows.EventCallback) (*flows.NLUClassification, bool, error) {
+	if input == "" {
+		return nil, true, errors.New("can't classify empty input, skipping classification")
+	}
+	if classifier == nil {
+		return nil, false, errors.Errorf("missing %s", a.Classifier.String())
+	}
+
+	nluSvc := run.Session().Engine().Services().NLU(run.Session(), classifier)
+	if nluSvc == nil {
+		return nil, false, errors.Errorf("no NLU service available for %s", a.Classifier)
+	}
+
+	classification, err := nluSvc.Classify(run.Session(), input, logEvent)
+	return classification, false, err
+}
+
+func (a *CallClassifierAction) saveSuccess(run flows.FlowRun, step flows.Step, input string, classification *flows.NLUClassification, logEvent flows.EventCallback) {
+	// result value is name of top ranked intent if there is one
+	value := ""
+	if len(classification.Intents) > 0 {
+		value = classification.Intents[0].Name
+	}
+	extra, _ := json.Marshal(classification)
+
+	a.saveResult(run, step, a.ResultName, value, classificationCategorySuccess, "", input, extra, logEvent)
```

Comment:
So for the result name, I take it we aren't doing `_whatever`? Or is that expected to be done by the editor? (in which case it isn't really a special case at all from the flow engine perspective just an editor convention?)

Reply:
So we need to talk about that and whether we want these results to be special in some way and maybe `_` indicates that.. but ultimately it's the editor that decides that. I just need to update Excellent if we want names that can start with `_`.

Reply:
I think we are all on the same page here. My plan is to have a single result name that the user enters "My Result" and then have a hidden implicitly prefixed result name for classification, `_classify_my_result` on the action. (or whatever prefix we agree on)

Reply:
I think we need to unpack what hidden means

Reply:
And I'm honestly not sure.. on the one hand the result.. seems like it would be best hidden because there's potential for confusing users with two similar results.. on the other hand we're saying this result contains the complete classification data - i.e. if you're the 1% user that wants all the extra gory details, use this result.. but they can't if they can't find it..

Reply:
Right, well there's two different hiddens here. I was referring to the user only being asked for one name on that dialog and us constructing an extra one for them behind the scenes. How we surfaces that (ie, in completion, on the dialog itself) is the next question. I'm thinking we start with fully hidden as a starting point.

Reply:
What does fully hidden mean? Not available in completion? We can't really have things be available in one place (used in expression for a router node) and not in others.

I would say just leave out `_` leading results from completion but if they type it explicitly in an expression it is there.

Reply:
And not included in results / exports etc..

Reply:
Maybe as extra credit one day we can make it appear in completions if you show your clue by typing the first `_`

Reply:
I like the idea of `_foo` results being completely hidden for now and becoming available if user indicates they know what they're doing by typing `_`.  I think that means from flow inspection pov.. we don't return those results so RP never knows about them and they don't appear on dashboards or in exports etc.

Reply:
Ok to clarify what we're agreeing to here..

- User enters `Foo` as the response name - that is the name of the router result and the action result is `_Foo Classification` which snakes to `_foo_classification`
- Editor won't display result names with `_` prefix in auto-completion for now.. (for extra credit.. typing `_` does add them to auto-completion)
- RP excludes these results from the results dashboard and exports

I got that right @ericnewcomer @nicpottier ?